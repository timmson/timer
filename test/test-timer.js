const React = require("react");
const renderer = require("react-test-renderer");

const Timer = require("../src/timer");

class MockAudio {

}

class MockWindow {

	addEventListener() {

	}

}

class Moment {

	locale(locale) {
		return (locale === "ru") ? this : null;
	}

	format(format) {
		return (format === "HH:mm:ss, dddd, DD.MM.YYYY Ð³.") ? "xxx" : null;
	}
}

function moment() {
	return new Moment();
}

xdescribe("Timer should", () => {

	test("create app", () => {
		const component = renderer.create(
			<Timer audio={new MockAudio()} window={new MockWindow()} moment={moment} variants={["01:00", "02:00"]}/>
		);
		expect(component.toJSON()).toMatchSnapshot();
		component.unmount();
	});

});
