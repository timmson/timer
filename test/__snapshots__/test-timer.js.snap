// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer should create app 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="variants col-sm"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        01:00
      </a>
    </div>
    <div
      className="variants col-sm"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        02:00
      </a>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="timer normal"
        onClick={[Function]}
      >
        05
        <span
          style={Object {}}
        >
          :
        </span>
        00
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col currentTime"
    >
      <a
        href="#"
        rel="noreferrer"
        target="_blank"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
      style={
        Object {
          "color": "#666",
          "textAlign": "right",
        }
      }
    >
      Версия 
      2022
       года. Ищете 
      <a
        href="./old/"
      >
        старую версию
      </a>
      ?
    </div>
  </div>
</div>
`;
